test/python/circuit/test_instructions.py

```diff
@@ -332,6 +332,11 @@ def test_modifying_copied_params_leaves_orig(self):
 
         self.assertEqual(inst.params, [0, 1, 2])
 
+    def test_instance_of_instruction(self):
+        qr = QuantumRegister(2)
+        qc = QuantumCircuit(qr)
+        self.assertRaises(CircuitError, qc.append(HGate, qr[:], []))
```

Comment:
You can also use `assertRaises` as context manager, like so, this should fix the errors.
```suggestion
        with self.assertRaises(CircuitError):
            qc.append(HGate, qr[:], [])
```

Reply:
why does the previous version fail? Shouldn't they be equivalent?